### PR TITLE
add validation and GlobalExceptionHandler

### PR DESCRIPTION
**Pull request by k1llzers**
add validation and GlobalExceptionHandler

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -28,47 +28,49 @@
...
         <dependency>
             <groupId>org.springframework.boot</groupId>
             <artifactId>spring-boot-starter-web</artifactId>
         </dependency>
-
         <dependency>
-            <groupId>org.springframework.boot</groupId>
-            <artifactId>spring-boot-starter-validation</artifactId>
+            <groupId>org.hibernate.validator</groupId>
+            <artifactId>hibernate-validator</artifactId>
```
Comment (Rommelua) [inline pom.xml]: better use starter

File: src/main/java/book/store/bookstore/controller/BookController.java [hunk 2]
```diff
@@ -29,7 +35,24 @@ public BookDto getBookById(@PathVariable Long id) {
...
+    @DeleteMapping("/{id}")
+    @ResponseStatus(HttpStatus.NO_CONTENT)
+    public void deleteBook(@PathVariable Long id) {
+        bookService.deleteById(id);
+    }
+
+    @GetMapping("/search")
+    public List<BookDto> search(BookSearchParameters bookSearchParameters) {
+        return bookService.search(bookSearchParameters);
+    }
```
Comment (Rommelua) [inline src/main/java/book/store/bookstore/controller/BookController.java]: How are these methods connected with the task?
image

**Review by Rommelua — CHANGES_REQUESTED** [review]
PR should not contain code from another task

File: src/main/java/book/store/bookstore/controller/BookController.java (hunk 2 above)
Comment (k1llzers) [reply src/main/java/book/store/bookstore/controller/BookController.java]: Yeah, I`m sorry, it was necessary to deal with github, now fixed it)

**Pull request merged**